client/my-sites/domains/domain-search/index.tsx

```diff
@@ -416,17 +450,17 @@ class DomainSearch extends Component {
 		return (
 			<Main className={ classes } wideLayout>
 				<QueryProductsList />
-				<QuerySiteDomains siteId={ this.props.selectedSiteId } />
+				<QuerySiteDomains siteId={ selectedSiteId } />
 				{ content }
 			</Main>
 		);
 	}
 }
 
 export default connect(
-	( state ) => {
+	( state: IAppState ) => {
 		const site = getSelectedSite( state );
-		const siteId = getSelectedSiteId( state );
+		const siteId = getSelectedSiteId( state ) ?? undefined;
```

Comment:
Treating potential `null` value as `undefined` to conform to the shape of the selectors we're calling below.